docs/en/toolkit/env_config_ref.md

```diff
@@ -0,0 +1,342 @@
+
+In contrast, if you are actively developing a custom bundle – say you’re working on writing a Toolkit app for a specific workflow in your studio, you may want to get it directly from a path on disk. In this case you’d use a descriptor of type `dev`, which might look like this: 
+
+```
+    location:
+        type: dev
+        path: /path/to/app
+```
+
+The `dev` descriptor has different settings from the `app_store` descriptor. While it can take other settings, it can be set up simply with a `path` setting pointing to where the app lives on disk.
+
+See [the Descriptor section of the Toolkit Core API docs](https://developer.shotgunsoftware.com/tk-core/descriptor.html) for details on all available descriptor types and their settings.
+
+## The apps block
+
+Apps are Toolkit’s user interfaces, and each can run independently of any others. You can choose which apps you want to use based on your pipeline needs, and the `apps` setting within an engine block is where you define which apps are available in a given engine.
+
+Here again is the `apps` setting from our example above:
+
+```
+engines:
+  tk-maya:
+    apps:
+      tk-multi-workfiles2:
+        location:
+          type: app_store
+          name: tk-multi-workfiles2
+          version: v0.11.8
+```
+
+You can see that we have a single app defined, the `tk-multi-workfiles2` app. It currently only has a single setting defined, its descriptor.
+
+If you wanted to make other apps available in the `tk-maya` engine in the `project` environment, you’d add them here. Let’s add the Panel, `tk-multi-shotgunpanel`, and the About app, `tk-multi-about`, to our engine. Our example `project.yml` file now looks like this:
+
+```
+engines:
+  tk-maya:
+    apps:
+      tk-multi-about:
+        location:
+          type: app_store
+          name: tk-multi-about
+          version: v0.2.8
+      tk-multi-shotgunpanel:
+        location:
+          type: app_store
+          name: tk-multi-shotgunpanel
+          version: v1.6.3
+      tk-multi-workfiles2:
+        location:
+          type: app_store
+          name: tk-multi-workfiles2
+          version: v0.11.8
+    location:
+        type: app_store
+        name: tk-maya
+        version: v0.9.4
+```
+
+There are a few important things to note at this time: 
+
+* The Default Config lists bundles in alphabetical order, and this example follows that convention.
+* The file is beginning to get long, and we haven’t even added any configuration settings yet. 
+* You might imagine that you’ll be using these same apps in other engines and other environments. For example, you’ll probably have all three of these apps – the Panel, the About app, and the Workfiles app – in different engines (say, Houdini, Nuke, or Photoshop), and in different environments (like `asset_step` or `shot_step`). Defining the same app settings in many places in your config means that when it comes time to make a change, you’ll have to make the modification in many places.
```

Comment:
Feel free to disregard as it's picky, but the workfiles app would need different settings for different software/engines so couldn't make use of the include sharing.

Reply:
i'm gonna disregard :). it's just an example, and it'd break up the flow from the apps mentioned throughout the example to bring in a new one now.